src/github.com/stellar/horizon/resource/account.go

```diff
@@ -71,3 +72,16 @@ func (this *Account) Populate(
 	this.Links.Data.PopulateTemplated()
 	return
 }
+
+// GetData returns decoded value for a given key. If the key does
+// not exist, empty slice will be returned.
+func (this *Account) GetData(key string) []byte {
+	data, exists := this.Data[key]
+	if !exists {
+		return nil
+	}
+
+	// We assume that horizon always returns valid base64 encoded string
+	decoded, _ := base64.StdEncoding.DecodeString(data)
```

Comment:
> If we could prudently check your whole list, I would argue we should. Indeed many of the examples you provide are validated by horizon when the data is coming from an outside source. This is financial software bartek!

OK, I think here's the fundamental conflict in your arguments or maybe my misunderstanding:

* If many of the things I listed are validated by horizon, then why can't we validate data value too (valid base64) and allow developers to decode it with assumption that it's correct? (so `GetData` without returning error). I understand that returning an error is another layer of defence but then:
* If we should check values returned by horizon in the client code, then why don't we already check if account ids, hashes, links and other things I listed are valid? We should also advice developers to check the full history of sender account when a payment arrives because horizon can send incorrect value because of a bug.

I think that there are three (non mutually exclusive) possible answers to this:
* Many validations in our clients code are missing.
* We should advise developers who use our clients to perform additional checks and don't fully trust horizon data. What we are not doing now.
* We should make developers lives easier and do not trust stellar-core data in horizon and check if data is valid and consistent, keep track of the current balance of each account etc.

No matter which one is correct (we should discuss it more on Slack), I think that the solution we can both agree on regarding this PR is having two methods:
* `MustGetData(key string) []byte` that will `panic` if the data value is not correct.
* `GetData(key string) []byte, error`